### DISCUSSION TITLE
Version pinning

### DISCUSSION THREAD

**Discussion by aerlaut** [fluxcd/flux2#3679] [Q&A]
I'm trying to pin the version of flux toolkit that is installed in the cluster. To do that, I use the command

    flux bootstrap github --version v0.40.1 ...

This command is put in a CI script which is run with other steps (e.g. cluster setup). I ran the command this morning and was faced with this error:

![image](https://user-images.githubusercontent.com/2862362/224316879-1631485d-65e7-44ad-959e-68cf7468cf57.png)

It seems that the version is not pinned. I was expecting the `--version` flag to pin Flux toolkit versions that are installed to the cluster. Did I understand something wrong?

**Comment by makkes** [comment]
Flux CLI version and target version need to match on the major and minor part of the version, meaning you need to install Flux CLI version "v0.40.*" to be able to install "v0.40.1".

It's usually simpler to install the CLI version that you're targeting instead of using the `--version` flag.
